packages/taiko-client/pkg/rpc/private_tx_pool.go

```diff
@@ -0,0 +1,84 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/go-resty/resty/v2"
+)
+
+const (
+	BlockPriceURL = "/gasprices/blockprices"
+)
+
+type BlocknativePrivateTxPool struct {
+	apiEndpoint string
+	apiKey      string
+	rpcClient   *ethclient.Client
```

Comment:
```suggestion
	rpcClient   *EthClient
```

Reply:
And `SuggestGasTipCap(ctx context.Context) (*big.Int, error)` in `BlocknativePrivateTxPool` so that we can keep `EthClient` clean.

Reply:
EthClient includes BlocknativePrivateTxPool, which would be better since we have different types of PrivateTxPool

Reply:
The `SuggestGasTipCap` function in `BlocknativePrivateTxPool` will Shadow the one in `EthClient`.

Reply:
> EthClient includes BlocknativePrivateTxPool, which would be better since we have different types of PrivateTxPool

It doesn't matter. It's similar to inheritance.

Reply:
That's we want

Reply:
@davidtaikocha What's your point?